### PR TITLE
isCurrentEvent API changes

### PR DESCRIPTION
**Pull request by TPGamesNL**
### Description
A certain pattern was found a lot throughout Skript's source code, and likely also in many addons: in a syntax element, return false if !isCurrentEvent(SomeEvent.class), and cast to SomeEvent at runtime. 
However, this pattern may lead to CCEs (see for example #4872), because there may be more than 1 currentEvent at parsetime. If the syntax element can handle only one of the currentEvents, but not another, this pattern will cause a CCE if the element is called with that other event type.

This should be fixed by adding runtime checks for the provided event, which is what most file changes in this PR are.
I've also added Javadocs to the isCurrentEvent methods, explaining the situation and what to do.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #4872

### CODE REVIEW

**Comment by TheLimeGlass** [comment]
Is this still in draft? Where are the methods mentioned? Why did you add instance checking of the events? Isn't the whole point of the isCurrentEvent to setup the getting method to instantly cast, as the init/isCurrentEvent does the instanceof checking?

I agree with the new methods, deprecate the old ones and default isCurrentEvent to isAnyCurrentEvent.

**Comment by TPGamesNL** [comment]
Talked with pickle in DMs about the changes, changed stuff around after that but forgot to edit the PR description.

PR description is now accurate again.

**Comment by TheLimeGlass** [comment]
I'm still confused, this isn't going to fix addons. The changes should be applied and I don't agree with the instanceof checking in the get method as the get method is called more than the init. If the issue was only because the isCurrentEvent didn't check all the events, then why can't that be updated to do so?

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/util/coll/CollectionUtils.java [hunk 1]
```diff
@@ -251,6 +251,17 @@ public static boolean containsAnySuperclass(final @Nullable Class<?>[] classes,
 		}
 		return false;
 	}
+
+	/**
+	 * @return whether the given object is an instance of any of the given classes
+	 */
+	public static boolean isAnyInstanceOf(Object object, Class<?>... classes) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/util/coll/CollectionUtils.java:258]: Is this still used?

**Review by APickledWalrus — APPROVED** [review]
Just the question, but LGTM

File: src/main/java/ch/njol/util/coll/CollectionUtils.java (hunk 1 above)
Comment (TPGamesNL) [reply src/main/java/ch/njol/util/coll/CollectionUtils.java:258]: Yea, it was made for the 'any' variant
image

many impls in Skript already have a check like `if (!(e instanceof X || e instanceof Y))`, and I didn't touch them.

**Pull request merged**